hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,120 @@
+
+## Propagated EVM Transaction Response Codes
+These response codes can appear for `ContractCall`, `ContractCreate`, and `EthereumTransaction`.
+It is possible for `ContractCreate` and `ContractCall` to create child transactions, and if one of them fails with a special revert or halt reason, the status is propagated to the top-level status of the transaction.
+`ContractCall` creates child transactions by executing logic that results in another call to a function/contract.
+`ContractCreate` creates child transactions by having logic containing calls in the constructor.
+
+- **CONTRACT_REVERT_EXECUTED**
+  - The default error for contract execution being reverted. This also happens when the REVERT opcode is executed in the contract.
+
+- **INVALID_SOLIDITY_ADDRESS**
+  - The system cannot find the user’s Solidity address.
+
+- **INVALID_ALIAS_KEY**
+  - An alias used in a `CryptoTransfer` transaction is not the serialization of a primitive Key message. This means a Key with a single Ed25519 or ECDSA (secp256k1) public key and no unknown protobuf fields.
+
+- **INVALID_SIGNATURE**
+  - The transaction signature is not valid.
+
+- **INSUFFICIENT_GAS**
+  - Not enough gas was supplied to execute the transaction.
+
+- **LOCAL_CALL_MODIFICATION_EXCEPTION**
+  - Local execution (query) is requested for a function that changes state.
+
+- **MAX_CHILD_RECORDS_EXCEEDED**
+  - A contract transaction tried to use more than the allowed number of child records, either through system contract records or internal contract creations.
+
+- **INVALID_CONTRACT_ID**
+  - The contract ID is invalid or does not exist.
+
+- **INVALID_FEE_SUBMITTED**
+  - The fee submitted is invalid.
+
+- **INSUFFICIENT_TX_FEE**
+  - The fee provided in the transaction is insufficient for this type of transaction.
+
+- **INSUFFICIENT_PAYER_BALANCE**
+  - The transaction payer could not afford a custom fee.
+
+- **CONTRACT_EXECUTION_EXCEPTION**
+  - This code is used for any contract execution-related error not handled by specific error codes listed above.
+  - *Example*: An error in Besu due to differences in supported EVM versions. For instance, if Besu supports up to the Shanghai version and the contract is compiled with the Cancun version containing opcode changes, this error would be thrown.
+
+Those are some possible, but very unlikely to happen, top-level status codes. 
+As for MAX_STORAGE_IN_PRICE_REGIME_HAS_BEEN_USED and MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED, they indicate too many contracts/entities in the entire system, so the caller can't do anything except wait until the network is fixed.
+MAX_CONTRACT_STORAGE_EXCEEDED applies to an individual contract but still is very hard to reach under normal operation.
+
+- **MAX_CONTRACT_STORAGE_EXCEEDED**
+  - Contract permanent storage exceeded the currently allowable limit.
+
+- **MAX_STORAGE_IN_PRICE_REGIME_HAS_BEEN_USED**
+  - All contract storage allocated to the current price regime has been consumed.
```

Comment:
(This one and the next one are examples of a response code explanation that doesn't really explain much.  What's a price regime anyway?  That's why you needed the sentences above to explain them.  This is in reference to my previous comment about fixing up the protobuf enum decl later.)